core/src/Kokkos_GraphNode.hpp

```diff
@@ -246,6 +247,51 @@ class GraphNodeRef {
                       std::forward<Functor>(functor));
   }
 
+#if defined(KOKKOS_ENABLE_CUDA) ||                                           \
+    (defined(KOKKOS_ENABLE_HIP) && defined(KOKKOS_IMPL_HIP_NATIVE_GRAPH)) || \
+    (defined(KOKKOS_ENABLE_SYCL) && defined(SYCL_EXT_ONEAPI_GRAPH))
+  /// The parameter @p exec can be used for multi-GPU graphs.
+  /// The captured nodes will be assigned to the device associated with @p exec.
+  /// @todo Support label.
+  template <class Functor>
```

Comment:
Should we constrain the `Functor` like we had done for the "then" node? E.g. something like

```
std::enable_if_t<std::is_invocable_r_v<
    void, const Kokkos::Impl::remove_cvref_t<Functor>, const ExecutionSpace&>>>
```

Reply:
:heavy_check_mark: